### PR TITLE
WIP: Add check for pvm lpar's activation state

### PR DESCRIPTION
**Pull request by Zaoliang**
we have sometimes issue with lpar activation at early stage. lpar could
be shutdown but it got activated by other worker. So the required lpar is not
available. Add the check and stop test immediately.
see https://progress.opensuse.org/issues/65663
verifiction:
https://openqa.suse.de/tests/4283842#step/bootloader/8

### CODE REVIEW

File: lib/bootloader_pvm.pm [hunk 1]
```diff
@@ -151,20 +151,21 @@ sub boot_hmc_pvm {
...
     my $hmc              = select_console 'powerhmc-ssh';
+    my $max_wait_time    = 6;
 
     # detach possibly attached terminals - might be left over
     type_string "rmvterm -m $hmc_machine_name --id $lpar_id && echo 'DONE'\n";
     assert_screen 'pvm-vterm-closed';
 
     # power off the machine if it's still running - and don't give it a 2nd chance
-    type_string "chsysstate -r lpar -m $hmc_machine_name -o shutdown --immed --id $lpar_id && echo 'LPAR SUCCESSFULLY SHUT DOWN'\n";
+    type_string "chlparstate -m $hmc_machine_name -o shutdown --id $lpar_id -w $max_wait_time && echo 'LPAR SUCCESSFULLY SHUT DOWN'\n";
```
Comment (foursixnine) [inline lib/bootloader_pvm.pm:161]: this change is not needed, as we need the lpar to be shut down as soon as possible.

File: lib/bootloader_pvm.pm [hunk 2]
```diff
@@ -151,20 +151,21 @@ sub boot_hmc_pvm {
...
     assert_screen [qw(pvm-poweroff-successful pvm-poweroff-not-running)], 180;
```
Comment (foursixnine) [inline lib/bootloader_pvm.pm:162]: Before this line, you can add a check, to verify that the lpar is really down, by filtering the output from lssyscfg:

```
for ((i=0; i<10; i++)); do lssyscfg -m $MACHINE  -r lpar --filter "lpar_ids=$LPAR" -F state | grep -q "Not Activated" && echo "LPAR IS DOWN" && break || echo "Waiting for lpar $LPAR to shutdown" && sleep 5; done;
```

and `assert_screen` for `LPAR IS DOWN` (you'd need to create a needle) and use the same solution to assert that the state of the lpar is "Running"

File: lib/bootloader_pvm.pm [hunk 3]
```diff
@@ -151,20 +151,21 @@ sub boot_hmc_pvm {
...
 
     # proceed with normal boot if is system already installed, use sms boot for installation
     my $bootmode = get_var('BOOT_HDD_IMAGE') ? "norm" : "sms";
     type_string "chsysstate -r lpar -m $hmc_machine_name -o on -b ${bootmode} --id $lpar_id && echo 'LPAR SUCCESSFULLY BOOTED'\n";
-    assert_screen "pvm-poweron-successful";
-
+    assert_screen [qw(pvm-poweron-successful lpar-still-activated)], 90;
+    die "lpar $lpar_id cannot be activated in $max_wait_time minutes. Please try to restart the test as workaround" if match_has_tag('lpar-still-activated');
```
Comment (foursixnine) [inline lib/bootloader_pvm.pm:168]: With the above changes, none of this is needed
```suggestion
```

**Comment by Zaoliang** [comment]
worked on another branch.

**Comment by foursixnine** [comment]
> worked on another branch.

you can always use: `git reset --hard ${REMOTE} ${COMMIT_OR_BRANCH}` to reset to a particular state and keep this PR open. Nevertheless, remember to link the new pr if you open one, with this one.

**Comment by Zaoliang** [comment]
sure, I got messed up my branch by rebase, thanks for your tips!

**Comment by Zaoliang** [comment]
let me try this :)

**Comment by Zaoliang** [comment]
new PR #10401

**Pull request closed without merging**